pkg/build/api/v1beta3/types.go

```diff
@@ -382,6 +386,21 @@ type SourceBuildStrategy struct {
 	ForcePull bool `json:"forcePull,omitempty" description:"forces the source build to pull the image if true"`
 }
 
+// BuildPostCommitSpec holds a build post commit hook specification.
+type BuildPostCommitSpec struct {
+	// Shell is an action that executes a shell script in a container
+	// running the build output image.
+	Shell string `json:"shell,omitempty" description:"shell script to be executed in a container running the build output image"`
```

Comment:
@smarterclayton the only argument I have is that with having concatenation in place we trade a little bit of user experience for those users who just want to write a chunk of "shell script", similar to what you do in Jenkins job definition. Making things like pipes, redirection, etc will be harder for them to do.
But I understand the concerns about security/quoting, so I can live with that :-)